src/plugins/charts/public/services/palettes/palettes.tsx

```diff
@@ -29,6 +29,10 @@ import { LegacyColorsService } from '../legacy_colors';
 import { MappedColors } from '../mapped_colors';
 import { workoutColorForValue } from './helpers';
 
+const mappedSeriesNames: Record<string, string> = {
```

Comment:
It looks like we're replicating the logic of the terms agg field formatter here. 

https://github.com/elastic/kibana/blob/main/src/plugins/data/common/search/aggs/utils/get_aggs_formats.ts#L130-L150

Is the root of the problem that some clients are calling the categorical color method with the formatted term and some with the unformatted term?

Reply:
Yes I know that the partition chart is sending it as __other__ for example. This logic exists in many places afaik.